### ISSUE TITLE
Stuck at Parsing JS files for IntelliSense

### ISSUE THREAD

**Issue by volkans80** [Microsoft/nodejstools#570]
Hi,
When open project, stuck at Parsing JS files for IntelliSense then highlighting etc not working.
Thanks,

**Comment by billti** [comment]
There's really not a lot we can do with the info provided. Some more info that would be helpful:

 - What version of Visual Studio are you using? 
 - What release on Node Tools for VS do you have installed? 
 - What version of Node/NPM do you have installed?
 - Are you seeing this consistently, or intermittently? 
 - Is this happening for all projects, or just one/some? 
 - If for just one, can you provide it for us to try a repro with?
 - If happening for all projects, do you have any other add-ins installed that may be causing issues (Web Essentials, ReSharper, etc...)? Are other VS project types all working fine?
 - Is VS frozen and needs to be killed? (I assume not if highlighting is then not working)
 - If VS becomes responsive (i.e. you can use the editor), how long is it "stuck"?
 - Is the CPU/memory high when this occurs?

Thanks.

**Comment by volkans80** [comment]
Sorry for that.

* Microsoft Visual Studio Professional 2015 Version 14.0.23107.0 D14REL
* Node.js Tools   1.1.31022.02
* Node.js Tools - Profiling   1.1.31022.02
* Node.js v0.10.31 NPM 1.4.23
* It is consistent.
* It is only my project. New projects are working.
* My project is new project but i added some packages.
  * body-parser@1.8.4
  * cookie-parser@1.3.5
  * debug@2.0.0
  * express@4.9.8
  * jade@1.6.0
  * jsonwebtoken@5.4.0
  * morgan@1.3.2
  * sequelize@3.12.0
  * serve-favicon@2.1.7
  * stylus@0.42.3
* It is not happening all projects.
* No VS is not frozen.
* VS is responsive
* CPU normal, Memory consumption is 600MB but stable

Thanks,

**Comment by billti** [comment]
Thanks for the update. If it's only your project and other projects are working, any chance we can get hold of your project to try and repro? (I can give you my private email address if it's not something you can post publicly).

**Comment by volkans80** [comment]
I tested it with untouched new project and it is still stuck.
I created new Basic Node.js Express 4 Application project.
Download link:
http://we.tl/4JMY8zz7t2

**Comment by mousetraps** [comment]
@volkans80 You mentioned that VS is responsive, so what do you mean by "stuck"? Can you take a screenshot of the point at which it's "stuck"?

**Comment by volkans80** [comment]
![image](https://cloud.githubusercontent.com/assets/3122625/11124696/f439b2ae-896e-11e5-9967-f0d894d8d1fb.png)

**Comment by mousetraps** [comment]
Thanks, looks like we'll need some more info... would you mind sending us a dump of devenv.exe when the issue occurs? The quickest way to create a dump is by right clicking the process in Task Manager, and selecting "create dump file". Process explorer, procdump, etc. also work. You can then zip it up and upload it somewhere (post a link here or send it to ntvshelp at microsoft com) Also be sure to try one of our latest releases: https://github.com/Microsoft/nodejstools/releases

**Comment by volkans80** [comment]
I installed new release but still stuck.
Dump file download link:
http://we.tl/KlCApJUD7p

**Comment by billti** [comment]
Per the initial report:

> ...then highlighting etc not working

I take it that is still occurring? Are you getting any intellisense features (such as completion drop-downs, formatting, etc..)? Or is it effectively like editing a plain-text file when this issue occurs?

**Comment by Ampakinetic** [comment]
I've got the latest dev build installed and have just had my HTML editor issue come back (as reported in https://github.com/Microsoft/nodejstools/issues/592) - I've put the dump file up on:

http://www.ediy.co.nz/temp/devenv.zip

The message at the bottom left of the editor is: "Waiting for parse to complete..." 

Please let me know when you've got the file so I can remove it.

**Comment by billti** [comment]
Thanks. Can you clarify on "my HTML editor issue". Are you editing an HTML file? I don't believe NTVS gets involved in editing if not editing a .js or .jade file.

**Comment by Ampakinetic** [comment]
I've explained it in https://github.com/Microsoft/nodejstools/issues/592 but it was closed as a dupe of this issue. I don't know if it's related or not.

**Comment by volkans80** [comment]
I just created new Express 4 project and everything is normal:
![image](https://cloud.githubusercontent.com/assets/3122625/11238369/c7dba338-8ded-11e5-9b5c-8284bf777bda.png)

I closed VS and reopen same project: No problem.

I closed VS and reopen same project again. I can use VS but message and progress bar stuck at bottom:
![image](https://cloud.githubusercontent.com/assets/3122625/11238430/2f458520-8dee-11e5-84e2-53e57dc75c0e.png)

**Comment by Ampakinetic** [comment]
This keeps happening and I lose work as the auto-save and recover only work sometimes.  For more background I've included the system info below.  I'm running the following on Windows 7 64-bit.  Could it be anything else I've installed? Did the dump I made earlier help with anything?  

I have mapped .dust files to use the HTML editor in the default encoding.  It's working fine one second and today I have had it freeze while I'm in the editor moving around the code, so it's not necessarily only an on-focus issue.  VS window Minimize/Maximise work fine, very fast screen re-paint, but everything else is locked waiting for a background thread to finish it seems.

```
Microsoft Visual Studio Community 2015
Version 14.0.23107.0 D14REL
Microsoft .NET Framework
Version 4.6.00081

Installed Version: Community

Visual Basic 2015   00322-20000-00000-AA439
Microsoft Visual Basic 2015

Visual C# 2015   00322-20000-00000-AA439
Microsoft Visual C# 2015

Visual C++ 2015   00322-20000-00000-AA439
Microsoft Visual C++ 2015

Application Insights Tools for Visual Studio Package   1.0
Application Insights Tools for Visual Studio

ASP.NET and Web Tools   14.0.20626.0
ASP.NET and Web Tools

ASP.NET Web Frameworks and Tools 2013   5.2.30624.0
For additional information, visit http://www.asp.net/

Common Azure Tools   1.5
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

Fix File Encoding   1.2
Prevents Visual Studio from adding BOM to UTF-8 files.

GenerateUnitTest   1.0
Generates unit test code for methods in classes under test.

GitHub.VisualStudio   1.0
A Visual Studio Extension that brings the GitHub Flow into Visual Studio.

Microsoft Azure Mobile Services Tools   1.4
Microsoft Azure Mobile Services Tools

Microsoft MI-Based Debugger   1.0
Provides support for connecting Visual Studio to MI compatible debuggers

Node.js Tools   1.1.31117.00
Adds support for developing and debugging Node.js apps in Visual Studio

Node.js Tools - Profiling   1.1.31117.00
Profiling support for Node.js projects.

NuGet Package Manager   3.0.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit http://docs.nuget.org/.

PreEmptive Analytics Visualizer   1.2
Microsoft Visual Studio extension to visualize aggregated summaries from the PreEmptive Analytics product.

Python Tools for Visual Studio   2.2.30718.00
Python Tools for Visual Studio provides IntelliSense, projects, templates, Interactive windows, and other support for Python developers.

Python Tools for Visual Studio - Django Integration   2.2.30718.00
Provides templates and integration for the Django web framework.

Python Tools for Visual Studio - Profiling Support   2.2.30718.00
Profiling support for Python projects.

SQL Server Data Tools   14.0.50616.0
Microsoft SQL Server Data Tools

Visual C++ for Cross Platform Mobile Development   1.0
Visual C++ for Cross Platform Mobile Development

Visual C++ for Cross Platform Mobile Development   1.0
Visual C++ for Cross Platform Mobile Development

Xamarin   3.11.785.0 (d8dbdd9)
Visual Studio extension to enable development for Xamarin.iOS and Xamarin.Android.

Xamarin.Android   5.1.5.1 (17843e3350f88de6656e39ce5dd379eb191d140d)
Visual Studio plugin to enable development for Xamarin.Android.

Xamarin.iOS   8.10.4.0 (c57a2c7d806a18cc56c1f9f3aaf91cb17b791d02)
Visual Studio extension to enable development for Xamarin.iOS.

Xamarin.iOS Unified Migration   1.0
Automated migration for Xamarin iOS Classic projects to Unified

Xamarin.TestCloud.Integration   1.0
Early preview of Xamarin Test Cloud integration
```

**Comment by mjbvz** [comment]
Please try downloading and using [NTVS 1.2](https://github.com/Microsoft/nodejstools/releases) Beta with ES6 IntelliSense. This is our new, modern intellisense engine and it offers much better performance and stability over our old, static analysis engine.

Give NTVS 1.2 Beta a try and please report any issues you run into. Thanks.
